test/integration/failover_stateful_test.lua

```diff
@@ -235,6 +239,86 @@ function g.test_leader_restart()
     end)
 end
 
+function g.test_leader_promote()
+    g.kingdom:connect_net_box()
+    helpers.retrying({}, function()
+        g.kingdom.net_box:call('get_coordinator')
+    end)
+
+    -------------------------------------------------------
+
+    local storage = g.cluster:server('storage-1')
+    local ok, err = storage.net_box:eval([[
+        return require('cartridge').failover_promote(...)
```

Comment:
Testing graphql API would be enough since it uses lua api internally. You better test possible errors: kingdom death, coordinator death

Reply:
Test graphql API and then return back to initial state (to avoid breaking next tests) via eval